recipes/nifti_clib/all/conanfile.py

```diff
@@ -0,0 +1,160 @@
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, replace_in_file, rm, rmdir
+from conan.tools.microsoft import is_msvc, is_msvc_static_runtime
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+
+class NiftiClibConan(ConanFile):
+    name = "nifti_clib"
+    description = "C libraries for NIFTI support"
+    license = "LicenseRef-LICENSE"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/NIFTI-Imaging/nifti_clib"
+    topics = ("image")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "use_nifti2": [True, False],
+        "use_cifti": [True, False],
+        "use_fslio": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "use_nifti2": True,
+        "use_cifti": False,  # seems to be beta?
+        "use_fslio": False  # Note in CMakeLists.txt: "If OFF, The copyright of this code is questionable for inclusion with nifti."
+    }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("zlib/1.3.1")
+        if self.options.use_cifti:
+            self.requires("expat/2.6.2")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["NIFTI_INSTALL_NO_DOCS"] = True
+        tc.variables["USE_NIFTI2_CODE"] = self.options.use_nifti2
+        tc.variables["USE_CIFTI_CODE"] = self.options.use_cifti
+        tc.variables["USE_FSL_CODE"] = self.options.use_fslio
+        tc.variables["NIFTI_BUILD_TESTING"] = False  # maybe this should be false? It downloads extra test data
+        if is_msvc(self):
+            tc.variables["USE_MSVC_RUNTIME_LIBRARY_DLL"] = not is_msvc_static_runtime(self)
+            tc.variables["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True
```

Comment:
This is something we're trying to move away from.

Unless the library’s build-system is already using it, we should avoid using this CMake functionality because it interacts poorly with C++ templates and breaks certain compiler features. Libraries that don't provide a `.def` file and do not use `__declspec()` declarations simply do not support shared builds for Windows, and we should not be providing configurations that are not supported by upstream


Maybe upstream could be conviced to add proper Windows support for this configuration?